Cogs.Publishers/UmlSchemaPublisher.cs

```diff
@@ -0,0 +1,142 @@
+        public string TargetDirectory { get; set; }
+        // boolean to determine whether to replace existing or not
+        public bool Overwrite { get; set; }
+
+        // list of all IDs created. Used to ensure no duplicates
+        private List<string> IdList = new List<string>();
+
+        public void Publish(CogsModel model)
+        {
+            if (CogsLocation == null)
+            {
+                throw new InvalidOperationException("Cogs location must be specified");
+            }
+            if (TargetDirectory == null)
+            {
+                throw new InvalidOperationException("Target directory must be specified");
+            }
+            if (Overwrite && Directory.Exists(TargetDirectory))
+            {
+                Directory.Delete(TargetDirectory, true);
+            }
+            // what about Overwrite being false and Directory.Exists(TargetDirectory))?
+
+            Directory.CreateDirectory(TargetDirectory);
+
+            XNamespace xmins = "http://www.omg.org/spec/XMI/20110701";
+            XNamespace umlns = "http://www.omg.org/spec/UML/20110701";
+            XElement xmodel = new XElement("packagedElement", new XAttribute(xmins + "type", "uml:Package"), 
+                new XAttribute(xmins + "id", "TestProject"), new XAttribute("name", "RestaurantMenu"));
+            // loop through classes
+            foreach (var item in model.ItemTypes)
+            {
+                // Create class
+                var newItem = new XElement(new XElement("packagedElement", new XAttribute(xmins + "type", "uml:Class"),
+                           new XAttribute(xmins + "id", createId(item.Name)),
+                           new XAttribute("name", item.Name)));
+                String extends = item.ExtendsTypeName;
+                // loop through properties of class and add to class
+                foreach(var property in item.Properties)
+                {
+                    var newProperty = new XElement("ownedAttribute", new XAttribute(xmins+ "type", "uml:Property"),
+                           new XAttribute(xmins + "id", createId(item.Name + "." + property.Name)),
+                           new XAttribute("name", property.Name));
+                    newProperty.Add(new XElement("type", new XAttribute(xmins + "idref", property.DataTypeName)));
+                    if(property.MinCardinality != null)
+                    {
+                        newProperty.Add(new XElement("lowerValue", new XAttribute(xmins + "type", "uml:LiteralInteger"),
+                            new XAttribute(xmins + "id", createId(item.Name + "." + property.Name + ".MinCardinality")),
+                            new XAttribute("value", property.MinCardinality)));
+                    }
+                    if (property.MaxCardinality != null)
+                    {
+                        // if max is "n" change to "*"
+                        if (property.MaxCardinality.Equals("n"))
+                        {
```

Comment:
Both the if and else are almost the same, lets combine them and set the value attribute's value via a inline condition or a temporary variable.